optimus/profiler/profiler.py

```diff
@@ -218,13 +167,138 @@ def to_file(self, path=None, output="html"):
+                                # The column name has been changed. Get the new name
+                                _renamed_columns.append(get_name(_col_name))
+                        # if a dict
+                        if is_dict(_col_names):
+                            for _col1, _col2 in _col_names.items():
+                                _renamed_columns.append({get_name(_col1): get_name(_col2)})
+
+                    else:
+                        _renamed_columns = _col_names
+                    return _renamed_columns
+
+                # New columns
+                new_columns = []
+
+                current_col_names = df.cols.names()
+                renamed_cols = match_renames(df.get_meta("transformations.columns"))
+                for current_col_name in current_col_names:
+                    if current_col_name not in renamed_cols:
+                        new_columns.append(current_col_name)
+
+                # Rename keys to match new names
+                profiler_columns = self.output_columns["columns"]
+                actions = df.get_meta("transformations.actions")
+                rename = actions.get("rename")
+                if rename:
+                    for k, v in actions["rename"].items():
+                        profiler_columns[v] = profiler_columns.pop(k)
+                        profiler_columns[v]["name"] = v
+
+                # Drop Keys
+                for col_names in match_actions_names(drop):
+                    profiler_columns.pop(col_names)
+
+                # Copy Keys
+                copy_columns = df.get_meta("transformations.actions.copy")
+                if copy_columns is not None:
+                    for source, target in copy_columns.items():
+                        profiler_columns[target] = profiler_columns[source].copy()
+                        profiler_columns[target]["name"] = target
+                    # Check is a new column is a copied column
+                    new_columns = list(set(new_columns) - set(copy_columns.values()))
+
+                # Actions applied to current columns
+
+                modified_columns = match_actions_names(Actions.list())
+                calculate_columns = modified_columns + new_columns
+
+                # Remove duplicated.
+                calculate_columns = list(set(calculate_columns))
+
+            elif not are_actions:
+                calculate_columns = None
+            # elif not is_cached:
+        else:
+            calculate_columns = columns
+
+        return calculate_columns
+
+    def is_cached(self):
+        """
+
+        :return:
+        """
+        return len(self.output_columns) > 0
 
     def dataset(self, df, columns="*", buckets=10, infer=False, relative_error=RELATIVE_ERROR, approx_count=True,
-                sample=10000, stats=True, format="dict", mismatch=None):
+                sample=10000, stats=True, format="dict", mismatch=None, advanced_stats=False):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Redefining built-in 'format'](https://app.codacy.com/app/Iron-AI/Optimus/pullRequest?prid=4494751)